src/overlays/ovl_i2/fox_me.c

```diff
@@ -1,97 +1,2527 @@
+            break;
+
+        case 2:
+            if (boss->dmgType == 1) {
+                boss->dmgType = 0;
+                boss->timer_054 = 5;
+                boss->swork[0] += 32;
+
+                func_8001D034(&boss->sfxPos, 0x3103605B, boss->unk_04A);
+
+                boss->unk_04A++;
+                if (boss->unk_04A >= 8) {
+                    boss->unk_04A = 7;
+                }
+
+                if (boss->swork[0] >= 0xFF) {
+                    boss->unk_04A = 0;
+                    Audio_KillSfx(&boss->sfxPos);
+                    if (boss->swork[1] == 0) {
+                        boss->swork[1]++;
+                        func_800BA808(gMsg_ID_3315, RCID_PEPPY);
+                    }
+
+                    boss->swork[0] = 0;
+
+                    Matrix_RotateZ(gCalcMatrix, (boss->obj.rot.z - 45.0f) * M_DTOR, 0);
+
+                    src.x = 0.0f;
+                    src.y = 156.0f;
+                    src.z = 930.0f;
+
+                    Matrix_MultVec3fNoTranslate(gCalcMatrix, &src, &dest);
+
+                    func_i2_80188298(boss->obj.pos.x + dest.x, boss->obj.pos.y + dest.y, boss->obj.pos.z + dest.z,
+                                     boss->obj.rot.z - 45.0f);
+                    func_i2_80188228(boss->obj.pos.x - dest.x, boss->obj.pos.y - dest.y, boss->obj.pos.z + dest.z,
+                                     boss->obj.rot.z - 45.0f);
+                    func_i2_80188228(boss->obj.pos.x + dest.y, boss->obj.pos.y - dest.x, boss->obj.pos.z + dest.z,
+                                     (boss->obj.rot.z - 45.0f) + 90.0f);
+                }
+            }
+            break;
+
+        case 3:
+            boss->timer_050 = 30;
+            boss->actionState = 4;
+            boss->timer_05C = 15;
+            boss->swork[0] = 0;
+            break;
+
+        case 4:
+            if (boss->timer_050 == 0) {
+                func_8007D2C8(boss->obj.pos.x, boss->obj.pos.y, boss->obj.pos.z + 500.0f, 30.0f);
+                func_8007BFFC(boss->obj.pos.x, boss->obj.pos.y, boss->obj.pos.z + 500.0f, 0.0f, 0.0f, 0.0f, 20.0f, 30);
+                boss->actionState = 5;
+
+                boss->info.unk_10 = 1000.0f;
+
+                func_8001A55C(&boss->sfxPos, 0x3103605B);
+                Audio_PlaySfx(0x3102405DU, &boss->sfxPos, 4U, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+            }
+            break;
+
+        case 5:
+            boss->obj.rot.z -= 3.0f;
+            if (boss->vel.z < 5.0f) {
+                boss->vel.z = boss->vel.z + 1.0f;
+            }
+            break;
+    }
+
+    if (boss->unk_078.z >= 360.0f) {
+        boss->unk_078.z = boss->unk_078.z - 360.0f;
+    }
+    if (boss->unk_078.z < 0.0f) {
+        boss->unk_078.z += 360.0f;
+    }
+}
+
+void func_i2_801887D0(Boss* boss) {
+    s32 i;
+
+    if (boss->timer_054 & 1) {
+        RCP_SetupDL_27();
+        gDPSetPrimColor(gMasterDisp++, 0, 0, 0, 255, 0, 255);
+    }
+
+    Matrix_RotateZ(gGfxMatrix, (M_PI / 4), 1);
```

Comment:
```suggestion
    Matrix_RotateZ(gGfxMatrix, M_PI / 4.0f, 1);
```
I've been using the float denominator. Do you think we should just use an integer?

Reply:
Yes, i think we should use an integer

Reply:
and the parenthesis make the calculation more enclosed, looks more clear to me